### PR TITLE
Create ProctoringAssignmentService

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: scripts/install/registerAssignmentService.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
```
Comment (jsconan) [inline scripts/install/registerAssignmentService.php]: wrong copyright year

File: model/ProctoringAssignmentService.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ * 
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```
Comment (jsconan) [inline model/ProctoringAssignmentService.php]: wrong copyright year

File: model/ProctoringAssignmentService.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        $result = false;
+        $class = new \core_kernel_classes_Class(EligibilityService::CLASS_URI);
+        $eligibilities = $class->searchInstances([
+            EligibilityService::PROPERTY_TESTTAKER_URI => $user->getIdentifier(),
+            EligibilityService::PROPERTY_DELIVERY_URI => $delivery->getUri(),
+        ]);
+
+        foreach ($eligibilities as $eligibility) {
+            /* @var \core_kernel_classes_Resource $eligibility*/
+            $testCenter = $eligibility->getOnePropertyValue(new \core_kernel_classes_Property(EligibilityService::PROPERTY_TESTCENTER_URI));
```
Comment (jsconan) [inline model/ProctoringAssignmentService.php]: Using this can lead to performances issue.
@jbout Can you confirm ?

**Comment by jsconan** [comment]
I created a test center, assigned a delivery and test takers, started an execution and then deleted the test center.

With your patch, the delivery is not visible anymore from the list of available deliveries. However the delivery is still visible in the list of in progress deliveries. As the related test center has been deleted, it is impossible to authorize the pending delivery...

**Comment by krampstudio** [comment]
@jsconan if the delivery is still in the "in progress" list, then the issue isn't solved

**Comment by hutnikau** [comment]
@jsconan the branch has been updated. Please review it again.

File: scripts/update/Updater.php [hunk 4]
```diff
@@ -283,6 +285,22 @@ public function update($initialVersion) {
...
+            $deliveryServerConfig = $deliveryExt->getConfig('deliveryServer');
+            $deliveryServerOptions = $deliveryServerConfig->getOptions();
+
+            $deliveryServerService = new DeliveryServerService($deliveryServerOptions);
+            $deliveryServerService->setServiceManager($this->getServiceManager());
+            $this->getServiceManager()->register(DeliveryServerService::CONFIG_ID, $deliveryServerService);
+
+            $this->setVersion('1.9.2');
+        }
     }
```
Comment (llecaque) [inline scripts/update/Updater.php]: @hutnikau @jsconan This is an update script too complex for a fix, this upgrade should have been a feature to 1.10

Comment (hutnikau) [inline scripts/update/Updater.php]: @llecaque should I create new pull request with increasing of version?

**Pull request merged**